cmd/cluster/transferowner.go

```diff
@@ -69,10 +247,63 @@ func (o *transferOwnerOptions) run() error {
 			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
 		}
 	}()
+	cluster, err := utils.GetClusterAnyStatus(ocm, o.clusterID)
+	o.cluster = cluster
+	o.clusterID = cluster.ID()
+
+	userDetails, err := ocm.AccountsMgmt().V1().Accounts().Account(o.newOwnerName).Get().Send()
+	userName, ok := userDetails.Body().GetUsername()
+	if !ok {
+		return fmt.Errorf("Failed to get username from new user id")
+	}
 
 	// Find and setup all resources that are needed
+	hiveCluster, err := utils.GetHiveCluster(o.clusterID)
+	hiveKubeCli, _, hivecClientset, err := getKubeConfigAndClient(hiveCluster.ID())
+	if err != nil {
+		return err
+	}
 
-	cluster, err := utils.GetCluster(ocm, o.clusterID)
+	response, err := ocm.AccountsMgmt().V1().AccessToken().Post().Impersonate(userName).Parameter("body", "/dev/null").Send()
+	if err != nil {
+		return fmt.Errorf("Can't send request: %v", err)
```

Comment:
Very nitpicky, but try to use %w for errors in formatting
https://stackoverflow.com/a/61287626